R/bcdc_search.R

```diff
@@ -176,9 +180,13 @@ bcdc_search <- function(..., license_id = NULL,
     }
   })
 
-  query <- paste0(
-    terms, ifelse(nzchar(terms), "+", ""),
-    paste(names(facets), facets, sep = ":", collapse = "+"))
+  # build query by collating the terms and any user supplied facets
+  # if there are no supplied facets (e.g., is_empty(facets) returns TRUE) just use terms)
+  query <- if (is_empty(facets)) {
+      paste0(terms)
+    } else {
+      paste0(terms, "+", paste(names(facets), paste0("\"", facets, "\""), sep = ":", collapse = "+"))
```

Comment:
Minor suggestion. If facets is empty then the facet name checking `lapply` statement on lines 175:181 shouldn't need to run, so it could go in this `else` block before `paste()`?

Reply:
Done!